src/BreakfastSerializer/JSONSerializer.php

```diff
@@ -0,0 +1,155 @@
+        $breadth    = array();
+
+        $object = $this->extractAndSetSingleDepthProperties($data, $breadth, $reflection, $object);
+        $object = $this->extractAndSetMultipleDepthProperties($breadth, $reflection, $object);
+
+        return $object;
+    }
+
+    /**
+     * @param array $data
+     * @param array $breadth
+     * @param \ReflectionClass $reflection
+     * @param mixed $object
+     * @return mixed
+     * @throws \Exception
+     * @todo refactor this out somewhere once we start implementing in new formats
+     */
+    protected function extractAndSetSingleDepthProperties(
+        array $data,
+        array& $breadth,
+        \ReflectionClass& $reflection,
+        $object
+    )
+    {
+        foreach($data as $key=>$value) {
+            if (true === is_array($value)) {
+                $breadth[$key] = $value;
+            } else {
+                try {
+                    $property = $reflection->getProperty($key);
+                    $property->setAccessible(true);
+                    $property->setValue($object, $value);
+                } catch (\ReflectionException $e) {
+                    //Non property so we ignore this
+                } catch (\Exception $e) {
+                    throw $e;
+                }
+            }
+        }
+
+        return $object;
+    }
+
+    /**
+     * @param array $breadth
+     * @param \ReflectionClass $reflection
+     * @param mixed $object
+     * @return mixed
+     * @throws \Exception
+     * @todo refactor this out somewhere once we start implementing in new formats
+     */
+    protected function extractAndSetMultipleDepthProperties(
+        array& $breadth,
+        \ReflectionClass& $reflection,
+        $object
+    )
+    {
+        $propertyData = array();
+
+        foreach($breadth as $key => $value) {
+            foreach ($value as $instance) {
+                $propertyData[] = $this->arrayToObject($instance);
+            }
+
+            try {
+                $property = $reflection->getProperty($key);
+                $property->setAccessible(true);
+                $property->setValue($object, $propertyData);
+            } catch (\ReflectionException $e) {
+                //Non property so we ignore this
+                throw $e;
+            }
+        }
+
+        return $object;
+    }
+
+    /**
+     * @param mixed $baseObject
+     * @param bool  $exposeClassName
+     * @return array
+     * @todo refactor this out somewhere once we start implementing in new formats
+     */
+    protected function objectToArray($baseObject, $exposeClassName = true)
+    {
+        $data = array();
+
+        $objAsArray = is_object($baseObject) ? (array)$baseObject : $baseObject;
+
+        if (true === SerializerFactory::canIterate($objAsArray)) {
+            foreach ($objAsArray as $key => $val) {
+                $val =
+                    (is_array($val) || is_object($val)) ? $this->objectToArray($val, $exposeClassName) : $val;
```

Comment:
I mean instead of using a Ternary. 

From

```php
$val = (is_array($val) || is_object($val)) ? $this->objectToArray($val, $exposeClassName) : $val;
```

To

```php
if (is_array($val) || is_object($val)) { 
    $var = $this->objectToArray($val, $exposeClassName);
}
```